packages/headless/src/features/facets/category-facet-set/category-facet-set-slice.ts

```diff
@@ -44,22 +44,29 @@ export const categoryFacetSetReducer = createReducer(
           return;
         }
 
-        state[facetId] = buildCategoryFacetRequest(options);
+        state[facetId] = {
+          request: buildCategoryFacetRequest(options),
+        };
       })
       .addCase(change.fulfilled, (_, action) => action.payload.categoryFacetSet)
       .addCase(restoreSearchParameters, (state, action) => {
         const cf = action.payload.cf || {};
         const facetIds = Object.keys(state);
 
         facetIds.forEach((id) => {
-          const request = state[id];
+          const request = state[id]?.request;
+
+          if (!request) {
+            return;
+          }
+
           const path = cf[id] || [];
           selectPath(request, path, request.numberOfValues);
         });
       })
       .addCase(updateCategoryFacetSortCriterion, (state, action) => {
         const {facetId, criterion} = action.payload;
-        const request = state[facetId];
+        const request = state[facetId]?.request;
 
         if (!request) {
           return;
```

Comment:
Would it be more user-friendly to throw errors when somebody is attempting to update/toggle/deselect/select a category facet that doesn't exist?

Reply:
I would prefer to not throw because some of the actions are reused in different facet reducers. For example, category and regular facets might dispatch the same action type, but ultimately only one reducer will handle it (the one containing the facet id). One option could be to refactor so that all actions are unique though,

Another reason is I do not see a failed sort criterion update due to an incorrect id as a unrecoverable error. I think we should only throw when the error is unrecoverable, or if it may corrupt the state (e.g. passing a string instead of a number).

With that said, I will think about this more, maybe there is a way to improve the debugging experience without throwing an error?